block-storage/src/main/scala/coop/rchain/blockstorage/util/io/RandomAccessIO.scala

```diff
@@ -1,44 +1,57 @@
 package coop.rchain.blockstorage.util.io
 
-import java.io.{FileNotFoundException, RandomAccessFile}
+import java.io.{EOFException, FileNotFoundException, RandomAccessFile}
 import java.nio.file.Path
 
 import cats.effect.Sync
 import cats.implicits._
-import IOError.IOErr
+import coop.rchain.blockstorage.util.io.IOError.RaiseIOError
 
 import scala.language.higherKinds
 
-final case class RandomAccessIO[F[_]: Sync] private (private val file: RandomAccessFile) {
-  def close(): F[IOErr[Unit]] =
+final case class RandomAccessIO[F[_]: Sync: RaiseIOError] private (
+    private val file: RandomAccessFile
+) {
+  def close: F[Unit] =
     handleIo(file.close(), ClosingFailed.apply)
 
-  def write(bytes: Array[Byte]): F[IOErr[Unit]] =
+  def write(bytes: Array[Byte]): F[Unit] =
     handleIo(file.write(bytes), ByteArrayWriteFailed.apply)
 
-  def writeInt(v: Int): F[IOErr[Unit]] =
+  def writeInt(v: Int): F[Unit] =
     handleIo(file.writeInt(v), IntWriteFailed.apply)
 
-  def readInt: F[IOErr[Int]] =
-    handleIo(file.readInt(), IntReadFailed.apply)
+  def readInt: F[Option[Int]] =
+    handleIoF(file.readInt().some, {
+      case _: EOFException => none[Int].pure[F]
+      case e               => RaiseIOError[F].raise[Option[Int]](IntReadFailed(e))
+    })
 
-  def readFully(buffer: Array[Byte]): F[IOErr[Unit]] =
-    handleIo(file.readFully(buffer), ByteArrayReadFailed.apply)
+  def readFully(buffer: Array[Byte]): F[Option[Unit]] =
+    handleIoF(file.readFully(buffer).some, {
+      case _: EOFException => none[Unit].pure[F]
```

Comment:
No, I am using Option here to distinguish EOF (so I can't stop reading). I can't return an empty array of size N (i.e. filled with zeros) because it will be indistingushiable from actually reading N zeros.